frontend/views/components/AvatarUpload.vue

```diff
@@ -0,0 +1,165 @@
+<template lang='pug'>
+  // TODO #658
+  form.c-avatar-form(@submit.prevent='')
+    .c-avatar-wrapper
+      label.c-avatar-label
+        avatar.c-avatar-img(
+          :src='form.picture'
+          ref='picture'
+        )
+        i18n.link.c-avatar-text Change avatar
+
+        input.sr-only(
+          type='file'
+          name='picture'
+          accept='image/*'
+          @change='fileChange($event.target.files)'
+          placeholder='http://'
+          data-test='avatar'
+        )
+      banner-scoped.c-formMsg(ref='formMsg' data-test='avatarMsg')
+</template>
+<script>
+import sbp from '~/shared/sbp.js'
+import imageUpload from '@utils/imageUpload.js'
+import Avatar from '@components/Avatar.vue'
+import BannerScoped from '@components/banners/BannerScoped.vue'
+import L from '@view-utils/translations.js'
+
+export default {
+  name: 'AvatarUpload',
+  props: {
+    variant: {
+      type: String,
+      validator: (value) => ['user', 'group'].includes(value),
+      required: true
+    }
+  },
+  components: {
+    Avatar,
+    BannerScoped
+  },
+  data () {
+    const picture = {
+      user: this.$store.getters.ourUserIdentityContract.attributes.picture,
+      group: this.$store.getters.groupSettings.groupPicture
+    }[this.variant]
```

Comment:
So this AvatarUpload is a neat way to DRY things, however, I see two issues with this specific design:

1. Using `data()` means these images will not be reactive and won't update for other clients when changed.
2. Tying the design of a generic component like this to Group Income's internals defeats the purpose of reusable components, and means that anytime we want to use this component for other types of image uploads, the component itself has to be modified. It should be designed in a way that is completely agnostic to internals, as if you were designing a pluggable library

Reply:
1. That's not true... It's being used the exact same logic as before, where we replace data value on `fileChange` using `setFromBlob`. I just doubled-check it and it works fine across clients.
2. I understand your concern and that was my first approach. The component accepting 2 props:
- `sbpParams` - Object `{ selector, contractID`, key }
- `picture` - String with the current picture.
But then I decided to centralize that info in one single place (computed property `sbpParams`). But anyway, I think that's indeed a better approach. I'll change it back.